scripts/zones/Western_Adoulin/npcs/Clemmar.lua

```diff
@@ -1,24 +1,24 @@
 -----------------------------------
 -- Area: Western Adoulin
---  NPC: Clemmar
+-- NPC: Clemmar
 -- Type: Standard NPC and Quest NPC
---  Involved with Quest: 'A Certain Substitute Patrolman'
 -- !pos -12 0 12 256
 -----------------------------------
 require("scripts/globals/missions");
 require("scripts/globals/quests");
 
-local quest_table =
+local quests =
 {
-    require("scripts/quests/adoulin/a_certain_substitute_patrolman")
+    {dsp.quest.log_id.ADOULIN, dsp.quest.id.adoulin.A_CERTAIN_SUBSTITUTE_PATROLMAN}
```

Comment:
i think i prefer the `require` method here - it lets you know exactly what the filename containing the quest data is (ie. i can yank inside the quotes and then open the file directly with that), and it already handles the whole "don't load it if it's already loaded" scenario (assuming it assigns it to a global in the same way mixins do)

Reply:
At the moment, `getQuest()` is what fetches a table from a quest file (which just returns its table without global assignment) - if it wasn't already in the global (`getQuest()` then stores the quest in the global if it had to fetch it from file). I'll make changes to have the quest files assign themselves to the global the same way mixins do.

After a quest file assigns itself to the global, should it return the table it just assigned (like mixins return their function), or just the Area/Quest pairs for the `involvedQuests` object to stash? Just the Area/Quest pair would be easiest (and at the point a quest is in the global, I don't know of additional value in returning the full table; all relevant functions just pull from the global based on Area/Quest index).

Reply:
to be honest, I think it should a) return the quest object, and b) the functions should use the quest object instead of the area/quest indices.  These tables are all passed by reference, so there shouldn't be any harm in passing the tables around.  I think it makes it easier to understand when the function is requesting an actual quest object instead of logid/questid